lib/pure/complex.nim

```diff
@@ -9,78 +9,150 @@
 
 
 
+
+
 ## This module implements complex numbers.
-{.push checks:off, line_dir:off, stack_trace:off, debugger:off.}
-# the user does not want to trace a part
-# of the standard library!
+## Complex numbers are currently implemented as generic on a 64-bit or 32-bit float.
 
+{.push checks: off, line_dir: off, stack_trace: off, debugger: off.}
+# the user does not want to trace a part of the standard library!
 
-import
-  math
 
-const
-  EPS = 1.0e-7 ## Epsilon used for float comparisons.
+import math
 
 type
-  Complex* = tuple[re, im: float]
-    ## a complex number, consisting of a real and an imaginary part
+  Complex*[T: SomeFloat] = tuple[re, im: T]
+    ## A complex number, consisting of a real and an imaginary part.
+  Complex128* = Complex[float64]
+    ## Alias for a pair of 64-bit floats.
+  Complex64* = Complex[float32]
+    ## Alias for a pair of 32-bit floats.
+
 
 const
-  im*: Complex = (re: 0.0, im: 1.0)
+  im*: Complex[float] = (re: 0.0, im: 1.0)
```

Comment:
The `im` should be deprecated and `Im` or `IM` should be used instead (like `PI` in `fenv` module)

Reply:
@mratsim the is no ``PI`` in ``fenv``, I just checked. But please no all caps identifiers, they are highly discouraged has they imply yelling and they are hard to type.